File: src/main/java/org/chousik/collection/validators/ValidatorMusicBandName.java
package org.chousik.collection.validators;

import org.chousik.exception.InvalidDataError;

/**
 * Валидатор названия музыкальной группы
 */
public class ValidatorMusicBandName implements IValidator<String> {
    /**
     * Проверяет строку на пустоту
     *
     * @param s строка
     * @throws InvalidDataError если строка пуста
     */
    @Override
    public void valide(String s) throws InvalidDataError {
        if (s.isBlank() | s.isEmpty()) {
            throw new InvalidDataError();
        }
    }
}

File: src/main/kotlin/collection/validators/ValidatorMusicBandName.kt
package org.chousik.collection.validators

import org.chousik.exception.InvalidDataError


class ValidatorMusicBandName : IValidator<String?> {

    override fun valide(t: String?) {
        if (t!!.isBlank() or t.isEmpty()) {
            throw InvalidDataError()
        }
    }
}